test/celeritas/em/UrbanMsc.test.cc

```diff
@@ -383,6 +384,176 @@ TEST_F(UrbanMscTest, step_conversion)
+                                                    0.21986948123587,
+                                                    8.8845468954555};
+        static double const expected_range_init[] = {1.0888680554349e-05,
+                                                     0.00053269138831022,
+                                                     0.015413789233736,
+                                                     0.21762788543933,
+                                                     15.553546812173};
+        static double const expected_range_fact[] = {0.2, 0.2, 0.2, 0.2, 0.2};
+        static double const expected_limit_min[]
+            = {1e-08, 1e-08, 1e-08, 1e-08, 1e-08};
+
+        auto result = sample(Algorithm::minimal, true);
+        EXPECT_VEC_SOFT_EQ(expected_mean_step, result.mean_step);
+        EXPECT_VEC_SOFT_EQ(expected_range_init, result.range_init);
+        EXPECT_VEC_EQ(expected_range_fact, result.range_fact);
+        EXPECT_VEC_EQ(expected_limit_min, result.limit_min);
+    }
+    {
+        // "Use safety" algorithm
+        static double const expected_mean_step[] = {2.1788745016992e-06,
+                                                    0.00010590181504835,
+                                                    0.003070788811936,
+                                                    0.15260432992607,
+                                                    8.8845468954555};
+        static double const expected_range_init[] = {5.4443402771744e-05,
+                                                     0.0026634569415511,
+                                                     0.07706894616868,
+                                                     1.0881394271966,
+                                                     77.767734060865};
+        static double const expected_range_fact[]
+            = {0.04, 0.04, 0.04, 0.13881394271966, 7.8067734060865};
+        static double const expected_limit_min[] = {1.9688399316472e-06,
+                                                    1.0522532283188e-05,
+                                                    3.1432398888924e-05,
+                                                    4.0583539826243e-05,
+                                                    3.6094312868035e-05};
+
+        auto result = sample(Algorithm::safety, true);
+        EXPECT_VEC_SOFT_EQ(expected_mean_step, result.mean_step);
+        EXPECT_VEC_SOFT_EQ(expected_range_init, result.range_init);
+        EXPECT_VEC_SOFT_EQ(expected_range_fact, result.range_fact);
+        EXPECT_VEC_SOFT_EQ(expected_limit_min, result.limit_min);
+    }
+    {
+        // "Use safety plus" algorithm
+        static double const expected_mean_step[] = {2.1788745016992e-06,
+                                                    0.00010590181504835,
+                                                    0.003070788811936,
+                                                    0.094489992160893,
```

Comment:
Interesting results!  Just for a note: it would be great that we can verify these results with respect to Geant4 (something like TestEm15) (a sanity check in the model level comparison to build our confidence). Yes, it has been one of my to-do list too - or may we leave such a validation for a summer student^^.